File: app/schemas/documents.py
from datetime import datetime

from pydantic import (
    BaseModel,
    ConfigDict,
    Field,
)


class DocumentSummary(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
    )

    document_id: str
    filename: str
    document_type: str | None = None
    file_size_bytes: int | None = Field(
        default=None,
        ge=0,
    )
    chunk_count: int = Field(
        ge=0,
    )
    uploaded_at: datetime | None = None


class DocumentDetail(
    DocumentSummary
):
    source: str | None = None
    file_hash: str | None = None


class DocumentListResponse(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
    )

    documents: list[DocumentSummary]
    count: int = Field(
        ge=0,
    )


class DocumentDeleteResponse(BaseModel):
    model_config = ConfigDict(
        extra="forbid",
    )

    document_id: str
    filename: str
    deleted: bool
    file_deleted: bool

File: frontend/src/types/documents.ts
export interface DocumentSummary {
  document_id: string;
  filename: string;
  document_type: string | null;
  file_size_bytes: number | null;
  chunk_count: number;
  uploaded_at: string | null;
}

export interface DocumentDetail
  extends DocumentSummary {
  source: string | null;
  file_hash: string | null;
}

export interface DocumentListResponse {
  documents: DocumentSummary[];
  count: number;
}

export interface DocumentDeleteResponse {
  document_id: string;
  filename: string;
  deleted: boolean;
  file_deleted: boolean;
}

export interface IngestResponse {
  duplicate: boolean;
  document_id: string | null;
  existing_document_id: string | null;
  filename: string;
  document_type: string | null;
  file_size_bytes: number | null;
  chunks_indexed: number | null;
  message: string;
  development_notice: string | null;
}

export interface UploadProgress {
  loaded: number;
  total: number;
  percentage: number;
}